core/src/main/java/com/datastax/oss/driver/api/core/data/CqlVector.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.api.core.data;
+
+import com.datastax.oss.driver.shaded.guava.common.base.Joiner;
+import com.datastax.oss.driver.shaded.guava.common.collect.ImmutableList;
+import com.datastax.oss.driver.shaded.guava.common.collect.Iterators;
+import java.util.Arrays;
+
+/** An n-dimensional vector defined in CQL */
```

Comment:
For the server type "n" is part of the type definition.  Should it be on the client as well?  Vector<float, 2> and Vector<float, 3> are different types.
You need to ensure that someone is not trying to bind a Vector<float, 3> into a Vector<float, 2>

Reply:
This is only the value type.  CqlVectorType actually defines the specific type of this vector, and there we include both subtype and dimension.  Also note that this is the type returned by CqlVectorCodec.getCqlType(), which in turn means that it's used as the comparison to determine whether or not a given DataType can be handled by the codec.  As a consequence the codec will reject any type which doesn't match it specifically.  You can see this tested in CqlVectorCodecTest.should_accept_vector_type_correct_dimension_only().